dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/IdentifiableObjectBundleHook.java

```diff
@@ -127,6 +126,29 @@ private void handleAttributeValues( IdentifiableObject identifiableObject, Objec
         }
     }
 
+    /**
+     * Make sure createdBy is immutable unless persistedObject.createdBy is
+     * null.
+     *
+     * @param object object from payload.
+     * @param persistedObject persistedObject.
+     * @param bundle The {@link ObjectBundle} of the importing service.
+     */
+    private void handleCreatedByProperty( IdentifiableObject object, IdentifiableObject persistedObject,
+        ObjectBundle bundle )
+    {
+        if ( persistedObject.getCreatedBy() == null )
+        {
+            object.setCreatedBy( object.getCreatedBy() != null ? object.getCreatedBy() : bundle.getUser() );
+        }
+        else if ( object.getCreatedBy() == null
```

Comment:
Just an idea, would it make sense for us to have a method `IdentifiableObjectUtils.equals( IdentifiableObject, IdentifiableObject )` with unit tests where the null logic is encapsulated? This will be similar to Java `Objects.equals`. Then we would not have to repeat all these null checks in application code/more compact code, and less risk of mistakes/nullpointer exceptions.

Reply:
fixed